server/src/main/java/io/crate/metadata/doc/DocIndexMetadata.java

```diff
@@ -460,6 +465,11 @@ private void internalExtractColumnDefinitions(@Nullable ColumnIdent columnIdent,
                 : "DataType used in table definition must have storage support: " + columnDataType;
             boolean docValuesDefault = storageSupport.getComputedDocValuesDefault(columnIndexType);
             boolean hasDocValues = Booleans.parseBoolean(columnProperties.getOrDefault(DOC_VALUES, docValuesDefault).toString());
+
+            // columnProperties.getOrDefault doesn't work here for OID values fitting into int.
+            // Jackson optimizes writes of small long values as stores them as ints:
+            long oid = Long.valueOf(columnProperties.getOrDefault("oid", COLUMN_OID_UNASSIGNED).toString());
```

Comment:
Can't we just cast it instead of converting it to a string and parsing it again?

Reply:
expression `long oid = (long) columnProperties.getOrDefault("oid", COLUMN_OID_UNASSIGNED)`
 fails with `Cannot cast 'java.lang.Integer' to 'long'`